space-framework/pkg/space-manager/provider.go

```diff
@@ -227,6 +236,59 @@ func (p *provider) processProviderWatchEvents() {
 	}
 }
 
+const (
+	SECRET_NS       = "default"
+	SECRET_DATA_KEY = "kubeconfig"
+)
+
+func (p *provider) createSpaceSecrets(space *spacev1alpha1apis.Space, spInfo spaceprovider.SpaceInfo) error {
+
+	var secret *v1.Secret
+	var secretName string
+
+	if spInfo.Config[spaceprovider.INCLUSTER] != "" {
+		secretName = "incluster-" + space.Name
+		secret = buildSecret(secretName, spInfo.Config[spaceprovider.INCLUSTER])
+		_, err := p.c.k8sClientset.CoreV1().Secrets(SECRET_NS).Create(p.c.ctx, secret, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		space.Status.InClusterSecretRef = &v1.SecretReference{
+			Name:      secretName,
+			Namespace: SECRET_NS,
+		}
+	}
+
+	if spInfo.Config[spaceprovider.EXTERNAL] != "" {
+		secretName = "external-" + space.Name
+		secret = buildSecret(secretName, spInfo.Config[spaceprovider.INCLUSTER])
+		_, err := p.c.k8sClientset.CoreV1().Secrets(SECRET_NS).Create(p.c.ctx, secret, metav1.CreateOptions{})
```

Comment:
I think first we need to check if the secret already exist, and than use create or update.

Reply:
same as above